docs/import/manage_disaster.js

```diff
@@ -628,39 +631,75 @@ function addDisaster() {
  * there is an existing disaster with the same details.
  * @param {string} disasterId of the form <year>-<name>
  * @param {Array<string>} states array of state (abbreviations)
- * @return {Promise<boolean>} returns true after successful write to firestore.
+ * @return {Promise<[null, boolean]>} returns when all ee folders have been
+ * created and firestore write has finished.
  */
 function writeNewDisaster(disasterId, states) {
   if (disasterData.has(disasterId)) {
     setStatus('Error: disaster with that name and year already exists.');
     return Promise.resolve(false);
   }
+  clearStatus();
   const currentData = createDisasterData(states);
   disasterData.set(disasterId, currentData);
 
-  const disasterPicker = $('#disaster-dropdown');
-  const disasterOptions = disasterPicker.children();
-  let added = false;
-  // We expect this recently created disaster to go near the top of the list, so
-  // do a linear scan down.
-  // Note: let's hope this tool isn't being used in the year 10000.
-  // Comment needed to quiet eslint.
-  disasterOptions.each(/* @this HTMLElement */ function() {
-    if ($(this).val() < disasterId) {
-      $(createOptionFrom(disasterId)).insertBefore($(this));
-      added = true;
-      return false;
-    }
+  const folderCreationPromises = [];
+  folderCreationPromises.push(
+      getCreateFolderPromise(eeLegacyPathPrefix + disasterId));
+  for (const state of states) {
+    folderCreationPromises.push(
+        getCreateFolderPromise(legacyStateDir + '/' + state));
+  }
+
+  const eePromisesResult = Promise.all(folderCreationPromises).then(() => {
+    const disasterPicker = $('#disaster-dropdown');
+    const disasterOptions = disasterPicker.children();
+    let added = false;
+    // We expect this recently created disaster to go near the top of the list,
+    // so do a linear scan down. Note: let's hope this tool isn't being used in
+    // the year 10000.
+    // Comment needed to quiet eslint.
+    disasterOptions.each(/* @this HTMLElement */ function() {
+      if ($(this).val() < disasterId) {
+        $(createOptionFrom(disasterId)).insertBefore($(this));
+        added = true;
+        return false;
+      }
+    });
+    if (!added) disasterPicker.append(createOptionFrom(disasterId));
+    toggleState(true);
+
+    disasterPicker.val(disasterId).trigger('change');
   });
-  if (!added) disasterPicker.append(createOptionFrom(disasterId));
-  toggleState(true);
 
-  disasterPicker.val(disasterId).trigger('change');
+  return Promise.all(
+      eePromisesResult,
+      disasterCollectionReference()
+          .doc(disasterId)
+          .set(currentData)
+          .then(() => true));
```

Comment:
subtle point, but you need to but the .then(() => true) after the Promise.all() closing parens, not inside, so it actually returns a boolean Promise as per the contract.

Reply:
Ah, I had actually changed the doc but this is better. Done.